addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/ExtensionOpenApiTest.java

```diff
@@ -0,0 +1,276 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.openapi;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import fi.iki.elonen.NanoHTTPD;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.charset.Charset;
+import java.util.List;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.support.HierarchyTraversalMode;
+import org.junit.platform.commons.support.ReflectionSupport;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.extension.spider.ExtensionSpider;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+public class ExtensionOpenApiTest extends AbstractServerTest {
+
+    private ExtensionOpenApi classUnderTest;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        classUnderTest = new ExtensionOpenApi();
+
+        Control control = mock(Control.class, withSettings().lenient());
+        setControlSingleton(control);
+
+        ExtensionLoader extensionLoader = mock(ExtensionLoader.class, withSettings().lenient());
+        when(control.getExtensionLoader()).thenReturn(extensionLoader);
+
+        given(extensionLoader.getExtension(ExtensionSpider.class))
+                .willReturn(mock(ExtensionSpider.class));
+    }
+
+    @Test
+    public void shouldImportFile() throws IOException {
+        // given
+        String test = "/PetStoreJson/";
+        String defnName = "defn.json";
+        this.nano.addHandler(new DefnServerHandler(test, defnName, "PetStore_defn.json"));
+        String fileContents =
+                getHtml(
+                        "PetStore_defn.json",
+                        new String[][] {{"PORT", String.valueOf(this.nano.getListeningPort())}});
+        File tmpFile = File.createTempFile("foo", "bar");
+        FileUtils.write(tmpFile, fileContents, Charset.defaultCharset());
+
+        // when
+        List<String> errors = classUnderTest.importOpenApiDefinition(tmpFile, false);
+
+        // then
+        assertThat("Should parse OK: " + errors, errors.isEmpty());
+    }
+
+    @Test
+    public void shouldImportMultiFileV3() {
+        // given
+        String test = "/PetStoreJson/";
+        String defnName = "defn.json";
+        String file = "v3/multi-file/api.yaml";
```

Comment:
@thc202 `multi-file/*` files are pulled in for testing functionality. This `api.yaml` enumerates the sibling files in the path as dependencies for the multi-part schema composition.

Reply:
I'm aware of that, I did not ask what the change was for but stated it was unrelated to DDN. If you open a PR saying that it does X and then it does W, X, Y, and Z, I can only assume the W, Y, and Z were not included on purpose (thus me asking to revert and do those unrelated changes in other PR).

If you want to add all changes in the same PR (which is fine), please, update the PR title/description accordingly. The unrelated changes should still be split into their own commits though.

Reply:
Thank you for the clarification on this. I've updated the changlelog to reflect the additional change that was added to help support the issues encountered while in development of the core functionality addition. Will I also need to update the PR comments to cover this?

Reply:
Thanks for the explanation and path forward. I've pushed the latest changes as two squashed commits representing the work done for the multi-file support and the additional commit for the DDN support.

Please let me know if I've missed anything.